test/gurka/test_route.cc

```diff
@@ -679,3 +679,42 @@ TEST(Standalone, OneWayIdToManyEdgeIds) {
     EXPECT_STREQ(e.what(), "No path could be found for input");
   }
 }
+TEST(Standalone, HonorAccessPropertyWhenConstructingRestriction) {
+  const std::string ascii_map = R"(
+        A<-----B<----D
+               ^
+               |
+        E----->F---->G
+)";
+
+  const gurka::ways ways = {
+      {"EF", {{"highway", "secondary"}, {"oneway", "yes"}, {"osm_id", "1"}}},
+      {"FG", {{"highway", "secondary"}, {"oneway", "yes"}, {"osm_id", "1"}}},
+      {"FB", {{"highway", "primary"}, {"oneway", "yes"}, {"osm_id", "2"}}},
+      {"DB", {{"highway", "primary"}, {"oneway", "yes"}, {"osm_id", "3"}}},
+      {"BA", {{"highway", "primary"}, {"oneway", "yes"}, {"osm_id", "3"}}},
+  };
+
+  const gurka::relations relations = {
+      {{
+           {gurka::way_member, "EF", "from"},
+           {gurka::way_member, "FB", "via"},
+           {gurka::way_member, "BA", "to"},
+       },
```

Comment:
`EF` and `FG` edges have the same `way_id`. So `GetGraphIds` return `GFBA` instead of `EFBA`. We must look at the access in my opinion. Let me know if the suggested solution is correct or not: it's the first that came to my mind